test/ctia/http/routes/scratchpad_test.clj

```diff
@@ -0,0 +1,287 @@
+          (is (= texts (:texts updated-scratchpad)))))
+
+      ;; bundle
+      (testing "POST /ctia/scratchpad/:id/bundle :add"
+        (let [new-bundle-entities {:malwares #{{:id "transient:616608f4-7658-49f1-8728-d9a3dde849d5"
+                                                :type "malware"
+                                                :schema_version ctim-schema-version
+                                                :name "TEST"
+                                                :labels ["malware"]}}}
+              response (post (str "ctia/scratchpad/" (:short-id scratchpad-id) "/bundle")
+                             :body {:operation :add
+                                    :bundle new-bundle-entities}
+                             :headers {"Authorization" "45c1f5e3f05d0"})
+              updated-scratchpad (:parsed-body response)]
+
+          (is (= 200 (:status response)))
+          (is (subset? (set (:malwares new-bundle-entities))
+                       (set (-> updated-scratchpad :bundle :malwares))))))
+
+      (testing "POST /ctia/scratchpad/:id/bundle :remove"
+        (let [deleted-bundle-entities {:malwares #{{:id "transient:616608f4-7658-49f1-8728-d9a3dde849d5"
+                                                    :type "malware"
+                                                    :schema_version ctim-schema-version
+                                                    :name "TEST"
+                                                    :labels ["malware"]}}}
+              response (post (str "ctia/scratchpad/" (:short-id scratchpad-id) "/bundle")
+                             :body {:operation :remove
+                                    :bundle deleted-bundle-entities}
+                             :headers {"Authorization" "45c1f5e3f05d0"})
+              updated-scratchpad (:parsed-body response)]
+          (is (= 200 (:status response)))
+          (is (not (subset? (set (:malwares deleted-bundle-entities))
+                            (set (-> updated-scratchpad :bundle :malwares)))))))
+
+      (testing "POST /ctia/scratchpad/:id/bundle :replace"
+        (let [bundle-entities {:malwares #{{:id "transient:616608f4-7658-49f1-8728-d9a3dde849d5"
+                                            :type "malware"
+                                            :schema_version ctim-schema-version
+                                            :name "TEST"
+                                            :labels ["malware"]}}}
+              response (post (str "ctia/scratchpad/" (:short-id scratchpad-id) "/bundle")
+                             :body {:operation :replace
+                                    :bundle bundle-entities}
+                             :headers {"Authorization" "45c1f5e3f05d0"})
+              updated-scratchpad (:parsed-body response)]
+          (is (= 200 (:status response)))
+          (is (= (:malwares bundle-entities)
+                 (-> updated-scratchpad :bundle :malwares)))))
```

Comment:
I suggest to add a test that check that only the `:malwares` value was updated.
For example using a GET just before.

~~~
(is (= old-bundle (into old-bundle bundle-entities)))
~~~

I'm not sure how :replace could take advantage of `deep-merging` instead of a simple `into` there. It would be nice to have an example for that specific case that will also help to test that we only replace a specific sub element of the scratchpad.